scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -82,41 +182,38 @@ def write(self, outfile=None):
             with open(outfile,'w') as xmlout:
                 xmlout.write(xmlstr)
 
-    def get_node(self, nodename, attributes=None, root=None, xpath=None):
+    def scan_node(self, nodename, attributes=None, root=None):
```

Comment:
Doing broad xpath searches is expensive. Rename these and only use them sparingly.